concourse/pipelines/guest-package-build.jsonnet

```diff
@@ -0,0 +1,1150 @@
+      ],
+    },
+    buildpackagejob {
+      package: 'guest-oslogin',
+      repos: ['deb9', 'deb10', 'deb11', 'deb11-arm64', 'el7', 'el8', 'el9'],
+      gcs_dir: 'oslogin',
+      uploads: [
+        uploadpackagetask {
+          package_paths: '{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin_((.:package-version))-g1+deb9_amd64.deb"}',
+          repo: 'gce-google-compute-engine-oslogin-stretch',
+          universe: 'cloud-apt',
+        },
+        uploadpackagetask {
+          package_paths: '{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin_((.:package-version))-g1+deb10_amd64.deb"}',
+          repo: 'gce-google-compute-engine-oslogin-buster',
+          universe: 'cloud-apt',
+        },
+        uploadpackagetask {
+          package_paths: '{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin_((.:package-version))-g1+deb11_amd64.deb"},{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin_((.:package-version))-g1+deb11_arm64.deb"}',
+          repo: 'gce-google-compute-engine-oslogin-bullseye',
+          universe: 'cloud-apt',
+        },
+        uploadpackagetask {
+          package_paths: '{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin-((.:package-version))-g1.el7.x86_64.rpm"}',
+          repo: 'gce-google-compute-engine-oslogin-el7',
+          universe: 'cloud-yum',
+        },
+        uploadpackagetask {
+          package_paths: '{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin-((.:package-version))-g1.el8.x86_64.rpm"}',
+          repo: 'gce-google-compute-engine-oslogin-el8',
+          universe: 'cloud-yum',
+        },
+        uploadpackagetask {
+          package_paths: '{"bucket":"gcp-guest-package-uploads","object":"oslogin/google-compute-engine-oslogin-((.:package-version))-g1.el9.x86_64.rpm"}',
+          repo: 'gce-google-compute-engine-oslogin-el9',
+          universe: 'cloud-yum',
+        },
+      ],
+    },
+    promotepackagejob {
+      package: 'guest-oslogin',
+      dest: 'stable',
+      promotions: [
+        promotepackagestabletask { universe: 'cloud-apt', repo: 'gce-google-compute-engine-oslogin-stretch' },
+        promotepackagestabletask { universe: 'cloud-apt', repo: 'gce-google-compute-engine-oslogin-buster' },
+        promotepackagestabletask { universe: 'cloud-apt', repo: 'gce-google-compute-engine-oslogin-bullseye' },
+        promotepackagestabletask { universe: 'cloud-yum', repo: 'gce-google-compute-engine-oslogin-el7' },
+        promotepackagestabletask { universe: 'cloud-yum', repo: 'gce-google-compute-engine-oslogin-el8' },
+        promotepackagestabletask { universe: 'cloud-yum', repo: 'gce-google-compute-engine-oslogin-el9' },
+      ],
+    },
+    buildpackagejob {
+      package: 'osconfig',
+      repos: ['deb10', 'deb11-arm64', 'el7', 'el8', 'el9', 'goo'],
```

Comment:
shouldn't this also include deb9 and deb11 as there are packages defined below per version?

Reply:
good catch. the error is actually in the uploads - we should be uploading the same package to multiple repos. updated.

Reply:
Ah yeah. Perhaps it is a good idea to call out this idea in in the definition of "repos". That sort of represents the build environment for the package more than the repo right?